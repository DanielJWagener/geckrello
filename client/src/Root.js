// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
