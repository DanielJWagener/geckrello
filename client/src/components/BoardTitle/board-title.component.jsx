import React, { Component } from "react";
import { connect } from "react-redux";

import { updateBoard } from "../../redux/boards/boards.actions";

import "./board-title.styles.scss";

export class BoardTitle extends Component {
  state = {
    inputText: "",
    titleText: ""
  };

  boardTitleInput = React.createRef();

  // Initially sets size of input and page title
  componentDidMount() {
    this.setInputSize(this.state.titleText.length);
    this.setPageTitle();
    this.setState({
      inputText: this.props.board.title || "",
      titleText: this.props.board.title || ""
    });
  }

  // Sets page title and input size every time state is updated
  componentDidUpdate() {
    this.setPageTitle();
    this.setInputSize();
  }

  // Updates state on user input
  inputHandler = event => {
    const inputText = event.target.value;

    this.setState({
      inputText: inputText
    });
  };

  // When user input hears Enter or Escape, calls checkEmptyInput
  keyDownHandler = event => {
    const newTitle = event.target.value;
    if (event.key === "Enter" || event.key === "Escape") {
      this.boardTitleInput.current.blur();
      this.checkEmptyInput(newTitle);
    }
  };

  // Calls checkEmptyInput when input loses focus
  onBlurHandler = event => {
    const newTitle = event.target.value;

    event.target.className = "board-title__input";

    this.checkEmptyInput(newTitle);
  };

  // Changes input styling on focus, and selects all input text
  onFocusHandler = event => {
    event.target.className = "board-title__input board-title__input--infocus";

    event.target.select();
  };

  // Changes input styling on mouse over
  onMouseOverHandler = event => {
    document.activeElement === event.target
      ? (event.target.className =
          "board-title__input board-title__input--infocus")
      : (event.target.className =
          "board-title__input board-title__input--hover");
  };

  // Changes input styling on mouse out
  onMouseOutHandler = event => {
    document.activeElement === event.target
      ? (event.target.className =
          "board-title__input board-title__input--infocus")
      : (event.target.className = "board-title__input");
  };

  // If user input is empty, resets value to most recently saved value of titleText
  // Otherwise, value of titleText set to current input value
  checkEmptyInput = newTitle => {
    if (newTitle === "") {
      this.setState({
        inputText: this.state.titleText
      });
    } else {
      this.setState({
        titleText: newTitle
      });
      this.props.updateBoard(this.props.board._id, { title: newTitle });
    }
  };

  // Changes input width dynamically
  setInputSize = () => {
    const input = document.querySelector("input");

    if (input) input.style.width = `${this.state.inputText.length * 10 + 30}px`;
  };

  // Changes page title dynamically
  setPageTitle = () => {
    document.title = `${this.state.inputText} - Geckrello, a Trello Clone`;
  };

  render() {
    return (
      <div>
        <input
          className="board-title__input"
          value={this.state.inputText}
          onChange={this.inputHandler}
          onKeyDown={this.keyDownHandler}
          tabIndex="0"
          onBlur={this.onBlurHandler}
          onFocus={this.onFocusHandler}
          onMouseOver={this.onMouseOverHandler}
          onMouseOut={this.onMouseOutHandler}
          ref={this.boardTitleInput}
        ></input>
      </div>
    );
  }
}

const mapStateToProps = ({ board }) => {
  return board;
};

export default connect(mapStateToProps, { updateBoard })(BoardTitle);
