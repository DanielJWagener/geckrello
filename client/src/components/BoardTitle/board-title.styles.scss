@import "../../sass/abstracts/_variables.scss";

.board-title {
  &__input {
    margin-left: 1rem;
    padding: 5px 0;
    grid-area: title;
    background-color: var(--color-primary);
    border: none;
    border-radius: 0.5rem;
    font-size: 2em;
    font-weight: 700;
    color: white;
    text-align: center;

    &--infocus {
      background-color: white;
      color: $default-font-color;
    }

    &--hover {
      background-color: var(--color-primary-hoverLight);
    }

    @media only screen and (max-width: 23.75em) {
      margin-left: 0.5rem;
    }
  }
}
