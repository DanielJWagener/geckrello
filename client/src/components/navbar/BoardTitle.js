import React, { Component } from "react";

class BoardTitle extends Component {
  state = {
    inputText: "Board Title",
    titleText: "Board Title"
  };

  // Initially sets size of input and page title
  componentDidMount() {
    this.setInputSize(this.state.titleText.length);
    this.setPageTitle();
  }

  // Sets page title and input size every time state is updated
  componentDidUpdate() {
    this.setPageTitle();
    this.setInputSize();
  }

  // Updates state on user input
  inputHandler = event => {
    const inputText = event.target.value;

    this.setState({
      inputText: inputText
    });
  };

  // When user input hears Enter or Escape, calls checkEmptyInput
  keyDownHandler = event => {
    const newTitle = event.target.value;

    if (event.key === "Enter" || event.key === "Escape") {
      this.checkEmptyInput(newTitle);
    }
  };

  // Calls checkEmptyInput when input loses focus
  onBlurHandler = event => {
    const newTitle = event.target.value;

    event.target.className = "navbar__input";

    this.checkEmptyInput(newTitle);
  };

  // Changes input styling on focus, and selects all input text
  onFocusHandler = event => {
    event.target.className = "navbar__input navbar__input--infocus";

    event.target.select();
  };

  // Changes input styling on mouse over
  onMouseOverHandler = event => {
    document.activeElement === event.target
      ? (event.target.className = "navbar__input navbar__input--infocus")
      : (event.target.className = "navbar__input navbar__input--hover");
  };

  // Changes input styling on mouse out
  onMouseOutHandler = event => {
    document.activeElement === event.target
      ? (event.target.className = "navbar__input navbar__input--infocus")
      : (event.target.className = "navbar__input");
  };

  // If user input is empty, resets value to most recently saved value of titleText
  // Otherwise, value of titleText set to current input value
  checkEmptyInput = newTitle => {
    if (newTitle === "") {
      this.setState({
        inputText: this.state.titleText
      });
    } else {
      this.setState({
        titleText: newTitle
      });
    }
  };

  // Changes input width dynamically
  setInputSize = () => {
    const input = document.querySelector("input");

    input.style.width = `${this.state.inputText.length * 10 + 30}px`;
  };

  // Changes page title dynamically
  setPageTitle = () => {
    document.title = `${this.state.inputText} - Geckorello, a Trello Front-End Clone`;
  };

  render() {
    return (
      <div>
        <input
          className="navbar__input"
          value={this.state.inputText}
          onChange={this.inputHandler}
          onKeyDown={this.keyDownHandler}
          onBlur={this.onBlurHandler}
          onFocus={this.onFocusHandler}
          onMouseOver={this.onMouseOverHandler}
          onMouseOut={this.onMouseOutHandler}
        ></input>
      </div>
    );
  }
}

export default BoardTitle;
