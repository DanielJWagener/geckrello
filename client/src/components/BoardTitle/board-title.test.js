import { shallow } from "enzyme";
import React from "react";
import BoardTitle from "./board-title.component";
import { BoardTitle as BoardTitlePure } from "./board-title.component";
import Root from "../../Root";

it("renders BoardTitle component", () => {
  expect(
    shallow(<BoardTitlePure board={{ title: "title" }} />).debug()
  ).toMatchSnapshot();
});
