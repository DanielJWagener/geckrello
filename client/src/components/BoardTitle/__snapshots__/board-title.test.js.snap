// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BoardTitle component 1`] = `
"<div>
  <input className=\\"board-title__input\\" value=\\"title\\" onChange={[Function]} onKeyDown={[Function]} onBlur={[Function]} onFocus={[Function]} onMouseOver={[Function]} onMouseOut={[Function]} />
</div>"
`;
